packages/core/src/api/blockManipulation/tables/tables.ts

```diff
@@ -224,78 +388,344 @@ export function resolveAbsoluteTableCellIndices(
+ *
+ * This method always starts the search for the row at the first column of the table.
+ *
+ * ```
+ * // Visual representation of a table
+ * | A | B | C |
+ * |   | D | E |
+ * | F | G | H |
+ * // "A" has a rowspan of 2
  *
- * @returns The row of the table.
+ * // getCellsAtRowHandle(0)
+ * // returns [
+ *  { row: 0, col: 0, cell: "A" },
+ *  { row: 0, col: 1, cell: "B" },
+ *  { row: 0, col: 2, cell: "C" },
+ * ]
+ *
+ * // getCellsAtColumnHandle(1)
+ * // returns [
+ *  { row: 1, col: 0, cell: "F" },
+ *  { row: 1, col: 1, cell: "G" },
+ *  { row: 1, col: 2, cell: "H" },
+ * ]
+ * ```
+ *
+ * As you can see, you may not be able to retrieve all nodes given a relative row index, as cells can span multiple rows.
+ *
+ * @returns All of the cells associated with the relative row of the table. (All cells that have the same relative row index)
  */
-export function getRow(
+export function getCellsAtRowHandle(
   block: BlockFromConfigNoChildren<DefaultBlockSchema["table"], any, any>,
-  relativeRowIndex: number
+  relativeRowIndex: RelativeCellIndices["row"]
 ) {
-  const occupancyGrid = getTableCellOccupancyGrid(block);
-  if (!occupancyGrid) {
-    throw new Error("Table is malformed");
-  }
-  const { width } = getDimensionsOfTable(block);
-  // First need to resolve the relative row index to an absolute row index
-  const { row: absoluteRow } = resolveRelativeTableCellIndices(
-    { row: relativeRowIndex, col: 0 },
-    block
-  );
+  try {
+    const occupancyGrid = getTableCellOccupancyGrid(block);
 
-  // Then for each column, get the cell at the absolute row index as a relative cell index
-  const cells = new Array(width).fill(false).map((_v, col) => {
-    return resolveAbsoluteTableCellIndices(
-      { row: absoluteRow, col },
-      block,
-      occupancyGrid
-    );
-  });
-  // Filter out duplicates based on row and col properties
-  return cells.filter((cell, index): cell is NonNullable<typeof cell> => {
-    return (
-      cell !== null &&
-      cells.findIndex((c) => c && c.row === cell.row && c.col === cell.col) ===
+    // First need to resolve the relative row index to an absolute row index
+    let absoluteRow = 0;
+
+    // Jump through the occupied cells ${relativeCellIndices.row} times to find the absolute row position
+    for (let i = 0; i < relativeRowIndex; i++) {
+      const cell = occupancyGrid[absoluteRow]?.[0];
+
+      if (!cell) {
+        // As a sanity check, if the cell is not occupied, we should throw an error
+        throw new Error(
+          `Unable to resolve relative table cell indices for table, cell at ${absoluteRow},0 is not occupied`
+        );
+      }
+
+      // Skip the cells that the rowspan takes up
+      absoluteRow += cell.rowspan;
+    }
+
+    // Then for each column, get the cell at the absolute row index as a relative cell index
+    const cells = new Array(occupancyGrid[0].length)
+      .fill(false)
+      .map((_v, col) => {
+        return getRelativeTableCellIndices(
```

Comment:
nitpicking on naming; I found it a bit confusing that this function returns the cells (and not just the indices). i.e.: I was confused how `getCellsAtRowHandle` could return cells while the "return" function returns indices